cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/type/impl/CassandraCalendarDateType.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.type.impl;
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.impl.AbstractGenericBasicType;
+import org.hibernate.type.descriptor.java.CalendarDateTypeDescriptor;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Cassandra doesn't do Calendar, but it does do Date, so convert them.
```

Comment:
Internal storage in cassandra 2.1 is actually long from epoch, just as inside java's Date.  The TZ handling doc is just about how the server interprets Strings to convert them. We're sending it longs instead, so handling the TZ is our problem.  Note that things are in a state of flux though - this code may need revision for https://issues.apache.org/jira/browse/CASSANDRA-7523